make pinning private github repository more intuitive

This url works:

```
opam pin add foo git+ssh://git@github.com/avsm/dev.git
```

but the more intuitive one is something like:

```
opam pin add foo git@github.com:avsm/dev
```

As works with `git clone`.  It took me quite a few iterations of the URL that worked before I realised that I forgot the additional `git@`

Sure:

```
$ opam pin add foo git@github.com:avsm/foo -vv
+ /usr/local/bin/ocamlc "-vnum"
- 4.06.0
[ERROR] Error getting source from ssh://git@github.com/avsm/foo
'opam pin add foo git@github.com:avsm/foo -vv' failed.
```

And

```
$ opam pin add foo git+ssh://git@github.com/avsm/foo.git -vv
+ /usr/local/bin/ocamlc "-vnum"
- 4.06.0
+ /usr/bin/git "init" (CWD=/Users/avsm/src/git/avsm/ocaml-dockerfile/_opam/.opam-switch/sources/foo)
- Initialized empty Git repository in /Users/avsm/src/git/avsm/ocaml-dockerfile/_opam/.opam-switch/sources/foo/.git/
+ /usr/bin/git "config" "--local" "fetch.prune" "false" (CWD=/Users/avsm/src/git/avsm/ocaml-dockerfile/_opam/.opam-switch/sources/foo)
+ /usr/bin/git "remote" "add" "origin" "ssh://git@github.com/avsm/foo.git" (CWD=/Users/avsm/src/git/avsm/ocaml-dockerfile/_opam/.opam-switch/sources/foo)
+ /usr/bin/git "remote" "set-url" "origin" "ssh://git@github.com/avsm/foo.git" (CWD=/Users/avsm/src/git/avsm/ocaml-dockerfile/_opam/.opam-switch/sources/foo)
+ /usr/bin/git "fetch" "-q" "ssh://git@github.com/avsm/foo.git" "--update-shallow" "+HEAD:refs/remotes/opam-ref" (CWD=/Users/avsm/src/git/avsm/ocaml-dockerfile/_opam/.opam-switch/sources/foo)
- ERROR: Repository not found.
- fatal: Could not read from remote repository.
-
- Please make sure you have the correct access rights
- and the repository exists.
+ /usr/bin/git "fetch" "-q" "ssh://git@github.com/avsm/foo.git" (CWD=/Users/avsm/src/git/avsm/ocaml-dockerfile/_opam/.opam-switch/sources/foo)
- ERROR: Repository not found.
- fatal: Could not read from remote repository.
-
- Please make sure you have the correct access rights
- and the repository exists.
[ERROR] Could not synchronize /Users/avsm/src/git/avsm/ocaml-dockerfile/_opam/.opam-switch/sources/foo from
        "git+ssh://git@github.com/avsm/foo.git":
        "/usr/bin/git fetch -q ssh://git@github.com/avsm/foo.git" exited with code 128 "ERROR: Repository not found."
[ERROR] Error getting source from git+ssh://git@github.com/avsm/foo.git
'opam pin add foo git+ssh://git@github.com/avsm/foo.git -vv' failed.
```

--- a/src/core/opamUrl.ml
+++ b/src/core/opamUrl.ml
@@ -116,8 +116,12 @@ let parse ?backend ?(handle_suffix=true) s =
         let of_suffix ~default =
           if not handle_suffix then default else
           match suffix with
-          | None -> default
-          | Some sf -> try vc_of_string sf with Failure _ -> default
+          | Some sf -> (try vc_of_string sf with Failure _ -> default)
+          | None ->
+            match OpamStd.String.cut_at path '@' with
+            | Some (user, _) ->
+              (try vc_of_string user with Failure _ -> default)
+            | None -> default
         in
         match transport with
         | None -> of_suffix ~default:`rsync

--- a/src/core/opamUrl.mli
+++ b/src/core/opamUrl.mli
@@ -32,7 +32,9 @@ type t = {
 
 (** Same as [of_string], but allows enforcing the expected backend, and may
     otherwise guess version control from the suffix by default (for e.g.
-    https://foo/bar.git) (this should be disabled when parsing from files) *)
+    https://foo/bar.git). (this should be disabled when parsing from files).
+    Note that [handle_suffix] also handles user-name in ssh addresses (e.g.
+    "ssh://git@github.com/...") *)
 val parse: ?backend:backend -> ?handle_suffix:bool -> string -> t
 
 include OpamStd.ABSTRACT with type t := t

